src/cpp/rtps/builtin/discovery/participant/PDPServerListener2.cpp

```diff
@@ -124,24 +124,85 @@ void PDPServerListener2::onNewCacheChangeAdded(
-            // {
-            //     logInfo(RTPS_PDP_LISTENER, "Malformed PDP payload received, ignoring: " << change->instanceHandle);
-            //     return;
-            // }
-
-            // Check whether the participant is a CLIENT or a SERVER
-            bool is_client = true;
+            /* Check PID_VENDOR_ID */
+            if (temp_participant_data_.m_VendorId != fastrtps::rtps::c_VendorId_eProsima)
+            {
+                logInfo(RTPS_PDP_LISTENER,
+                        "DATA(p|Up) from different vendor is not supported for Discover-Server operation");
+                return;
+            }
+
             fastrtps::ParameterPropertyList_t properties = temp_participant_data_.m_properties;
-            for (auto property : properties)
+
+            /* Check DS_VERSION */
+            auto ds_version = std::find_if(
+                properties.begin(),
+                properties.end(),
+                [](const dds::ParameterProperty_t& property)
+                {
+                    return property.first() == dds::parameter_property_ds_version;
+                });
+
+            if (ds_version != properties.end())
+            {
+                if (std::stof(ds_version->second()) < 1.0)
+                {
+                    logError(RTPS_PDP_LISTENER, "Minimum " << dds::parameter_property_ds_version
+                                                           << " is 1.0, found: " << ds_version->second());
+                    return;
+                }
+                logInfo(RTPS_PDP_LISTENER, "Participant" << dds::parameter_property_ds_version << ": "
+                                                         << ds_version->second());
+            }
+            else
+            {
+                logWarning(RTPS_PDP_LISTENER, dds::parameter_property_ds_version << " is not set. Assuming 1.0");
+            }
+
+            /* Check PARTICIPANT_TYPE */
+            bool is_client = true;
+            auto participant_type = std::find_if(
+                properties.begin(),
+                properties.end(),
+                [](const dds::ParameterProperty_t& property)
+                {
+                    return property.first() == dds::parameter_property_participant_type;
+                });
+
+            if (participant_type != properties.end())
             {
-                // If property PID_PERSISTENCE_GUID is set, then the participant is not a CLIENT
-                if (property.first() == "PID_PERSISTENCE_GUID")
+                if (participant_type->second() == ParticipantType::SERVER ||
+                        participant_type->second() == ParticipantType::BACKUP)
                 {
                     is_client = false;
-                    break;
                 }
+                else if (participant_type->second() != ParticipantType::CLIENT)
+                {
+                    logError(RTPS_PDP_LISTENER, "Wrong " << dds::parameter_property_participant_type << ": "
```

Comment:
Are we sure this should be an error? What if later on a participant sends `ParticipantType::SIMPLE` ?